src/pages/main/components/selected-dataset/build-flow/components/cards/description-card/description-card.tsx

```diff
@@ -0,0 +1,169 @@
+import styles from './description-card.module.css'
+
+import { useEffect, useState } from 'react'
+import { useHistory } from 'react-router'
+import cn from 'classnames'
+
+import { ExploreGenomeTypes } from '@core/enum/explore-genome-types-enum'
+import { getApiUrl } from '@core/get-api-url'
+import { t } from '@i18n'
+import { datasetStore } from '@store/dataset'
+import dirinfoStore from '@store/dirinfo'
+import filterStore from '@store/filter'
+import { Button } from '@ui/button'
+import { Card } from '@ui/card'
+import { Icon } from '@ui/icon'
+import { GlbPagesNames } from '@glb/glb-names'
+import { showToast } from '@utils/notifications/showToast'
+import {
+  exploreCandidateOptionsList,
+  optionsForOpenButton,
+} from '../../wizard/wizard.data'
+import { ICardProps } from '../../wizard/wizard.interface'
+import wizardStore from '../../wizard/wizard.store'
+import { getNextPageData, memorizeLocation } from '../../wizard/wizard.utils'
+import { CardTitleWithEdit } from '../components/card-edit-title'
+import { CardRadioList } from '../components/card-radio-list'
+
+const descriptions: { [key: string]: string } = {}
+let typingTimer: ReturnType<typeof setTimeout>
+
+export const DescriptionCard = (props: ICardProps) => {
+  const {
+    title,
+    id,
+    selectedValue,
+    continueDisabled,
+    contentDisabled,
+    editDisabled,
+  } = props
+  const history = useHistory()
+  const ds = title || datasetStore.datasetName
+
+  const note = descriptions[ds] || dirinfoStore.dirInfoData?.dsDict[ds]?.note
+  const isOpenButton = optionsForOpenButton.includes(selectedValue)
+  const [isEditMode, setEditMode] = useState(false)
+  const [isTyping, setTyping] = useState(false)
```

Comment:
move all data/state manipulations to description-card.store.ts
Ideally the only things that should be left here are: useEffect (because it reacts directly to the component props) and history operations (because that's what view components do). Stuff like editing, typing, description storage, et cetera should be moved to the store.